apps/nestjs-wallet/src/vc-api/exchanges/entities/transaction.entity.ts

```diff
@@ -75,6 +76,54 @@ export class TransactionEntity {
   @Column('simple-json')
   callback: CallbackConfiguration[];
 
+  private verifyVpRequestTypeDidAuth(presentation: VerifiablePresentation): void {
+    // https://w3c-ccg.github.io/vp-request-spec/#did-authentication-request
+    if (!presentation.holder) throw Error('Presentation holder is required for didAuth query');
+  }
+
+  private verifyVpRequestTypePresentationDefinition(
+    presentation: VerifiablePresentation,
+    credentialQuery: Array<{ presentationDefinition: IPresentationDefinition }>
+  ): void {
+    // https://identity.foundation/presentation-exchange/#presentation-definition
+    const vcSubjects = new Set([...presentation.verifiableCredential.map((vc) => vc.credentialSubject.id)]);
+    if (vcSubjects.size > 1) throw Error('Credential subjects are not the same');
+
+    const pex: PEX = new PEX();
+
+    credentialQuery.forEach(({ presentationDefinition }) => {
+      const { errors } = pex.evaluatePresentation(
+        presentationDefinition,
+        presentation as unknown as IPresentation
+      );
+
+      const combinedErrorMessage = errors.map((error) => error.message || 'unknown error').join(', ');
+
+      if (errors.length > 0) {
+        throw Error(`Presentation definition validation failed: ${combinedErrorMessage}`);
+      }
+    });
+  }
+
+  private validatePresentation(presentation: VerifiablePresentation): void {
+    // Common checking
+    if (presentation.proof.challenge !== this.vpRequest.challenge) throw Error('Challenge does not match');
```

Comment:
I don't think it makes sense to check this here actually. I think that the `challenge` doesn't necessarily have to be in the proof. For instance, an EIP712 signature doesn't include the `challenge` https://w3c-ccg.github.io/ethereum-eip712-signature-2021-spec/#example-3
```suggestion
```

Reply:
So where is the challenge in EIP712 signature? Maybe we should check it during verification?

Reply:
Let's leave this check for now. I think my previous comment might have been wrong. I think that maybe the `challenge` could be in an EIP712 proof, even though I haven't seen any example of it yet. Best I can tell at this point from the spruce/ssi code is that the proof in the signature option does indeed make it into the signed payload (https://github.com/spruceid/ssi/blob/f4fc47825a79aba2306ff78c679da3c9a457e755/src/ldp.rs#L465)
However, I think we have a bit of work to figure out what this looks like for EIP712 as the proof that we have thus far for an EIP712 VP proof doesn't contain a challenge: https://github.com/energywebfoundation/iam-client-lib/blob/3680858331dcc72b437a7c3bed8b70ebd1a85e44/e2e/verifiableCredentials.service.e2e.ts#L158

Reply:
I can't find a challenge in JSON-LD context https://w3c-ccg.github.io/ethereum-eip712-signature-2021-spec/contexts/v1/